src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,56 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
+
+  getInfo() {
+    const info = {
+      name: this.name,
+      health: this.health,
+    };
+
+    if (this.hidden !== undefined) {
+      info.hidden = this.hidden;
+    }
+
+    return info;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey instanceof Herbivore && !prey.hidden) {
+      prey.health -= 50;
+      prey.checkHealth();
```

Comment:
After decreasing the health of the prey, consider filtering the `Animal.alive` array directly in the `bite` method to remove any animals with health less than or equal to 0. This would eliminate the need for the `checkHealth` and `die` methods, streamlining the class design.